src/bytes.rs

```diff
@@ -81,3 +145,37 @@ impl<Inner: Buf> PrefixVarIntBuf for Inner {
         }
     }
 }
+
+/// An iterator over `PrefixVarInt` values in a `Buf`.
+pub struct PrefixVarIntIter<'a, B, PV> {
+    buf: &'a mut B,
+    _marker: std::marker::PhantomData<PV>,
+}
+
+// new method
+impl<'a, B, PV> PrefixVarIntIter<'a, B, PV> {
+    /// Creates a new `PrefixVarIntIter`.
+    pub fn new(buf: &'a mut B) -> Self {
+        Self {
+            buf,
+            _marker: std::marker::PhantomData,
+        }
+    }
+}
+
+impl<'a, B, PV> Iterator for PrefixVarIntIter<'a, B, PV>
+where
+    B: Buf,
+    PV: PrefixVarInt,
+{
+    type Item = Result<PV, DecodeError>;
+
+    #[inline]
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.buf.has_remaining() {
+            Some(self.buf.get_prefix_varint())
+        } else {
+            None
+        }
+    }
```

Comment:
would you please implement `size_hint()` as well? something like:
```
fn size_hint(&self) -> (usize, Option(usize)) {
    (self.remaining() / MAX_LEN, Some(self.remaining())
}
```

Reply:
Good point, implemented.